### PR TITLE
[SCHEMA] Various fixes on selectors and expressions

### PR DESCRIPTION
**Pull request by effigies**
The big one is 452a0b7, which will resolve https://github.com/bids-standard/bids-examples/issues/398. Otherwise, I was mostly clearing up redundant `modality` selectors, which led to seeing some insufficiently precise selectors.

### CODE REVIEW

File: src/schema/rules/tabular_data/pet.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ Blood:
       level_addendum: required if `MetaboliteAvail` is `true`
     metabolite_polar_fraction:
       level: optional
-      level_addendum: required if `MetaboliteAvail` is `true`
+      level_addendum: recommended if `MetaboliteAvail` is `true`
```
[the remark this answers is not in the document]
Comment (effigies) [reply src/schema/rules/tabular_data/pet.yaml:17]: It was recommended in 1.6.0, and the actual rule has remained recommended (L44):

https://github.com/bids-standard/bids-specification/blob/60c173d3a576f45a4241f1185a5819de892ec10f/src/schema/rules/tabular_data/pet.yaml#L35-L44

This was just a typo when creating this table.

File: src/schema/rules/sidecars/fmap.yaml [hunk 2]
```diff
@@ -8,15 +8,16 @@
...
     B0FieldIdentifier: recommended
 
 MRIFieldmapIntendedFor:
   selectors:
-    - modality == "mri"
     - datatype == "fmap"
+    - match(extension, '\.nii(\.gz)?$')
   fields:
     IntendedFor:
       level: optional
```
[the remark this answers is not in the document]
Comment (effigies) [reply src/schema/rules/sidecars/fmap.yaml:23]: Sure.

File: src/schema/rules/tabular_data/pet.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tsalo) [reply src/schema/rules/tabular_data/pet.yaml:17]: Ah, thanks!

**Review by tsalo — APPROVED** [review]
LGTM

**Comment by effigies** [comment]
Had to [partially undo](https://github.com/bids-standard/bids-specification/compare/ce1dc74eee00a253e6be98e1b76ba2febffef12b..7cd74a0fd9e817c535434118340d17b925587839) that last one. Task metadata is split across a bunch of modalities.

**Pull request merged**